src/kagglehub/gcs_upload.py

```diff
@@ -66,16 +71,26 @@ def _upload_blob(file_path: str, model_type: str):
         token_exception = "'token' field missing from response"
         raise BackendError(token_exception)
 
+    headers = {"Content-Type": "application/octet-stream"}
+
     with open(file_path, "rb") as f:
-        headers = {"Content-Type": "application/octet-stream"}
-        # TODO(b/312511716): add resumable upload
-        # TODO(b/312511716): Implement a progress bar for the upload using tqdm
-        gcs_response = requests.put(response["createUrl"], data=f, headers=headers, timeout=600, stream=True)
-        if gcs_response.status_code != 200:  # noqa: PLR2004
-            gcs_exception = (
-                f"Failed to upload to GCS. Status: {gcs_response.status_code}, Response: {gcs_response.text}"
-            )
-            raise Exception(gcs_exception)
+        for i in tqdm(range(0, file_size, CHUNK_SIZE), desc="Uploading"):
+            f.seek(i)  # Seek to the start of the chunk
```

Comment:
You don't need to seek for each chunk. When you call `f.read()`, the pointers moves automatically. 

My previous comment is about the resumable upload part. Where does it retry from where it left of after a failure? 

Let's say you are trying to upload a 10MB file but it fails after 1 MB. It should only try to re-upload the last 9MB no?

Reply:
Basically, you need to do step 2 & 3: https://cloud.google.com/storage/docs/performing-resumable-uploads#status-check